keyboards/z12/z12.c

```diff
@@ -0,0 +1,17 @@
+/* Copyright 2020 Sergi Meseguer <zigotica@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "z12.h"
```

Comment:
```suggestion
#include "z12.h"

#ifdef ENCODER_ENABLE
void encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) { // LEFT
        // Scroll
        if (clockwise) {
            tap_code_delay(KC_PGDN, 10);
        } else {
            tap_code_delay(KC_PGUP, 10);
        }
    } else { // RIGHT
        // Volume control.
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
}
#endif
```

Reply:
Hi @drashna thank you for taking the time to reviewing my z12 micropad code. I merged all your suggestions. 

I know it is probably explained somewhere, but couldn't find it. Why is it better to have the encoder behavior in the z12.c file instead of the keymap? Wouldn't it be better to let each user have their own encoder behavior? 

Thank you again

Reply:
It allows anyone compiling with configurator to have a working encoder. The function can still be overridden using the user version at a keymap level.

But shouldn't this be `bool encoder_update_kb(uint8_t index, bool clockwise)` with a return at the end?

Reply:
Ah, the configurator! THAT makes sense. Thank you very much.

Reply:
Also updated the bool/return @daskygit thank you
My question now is, on @drashna suggestion, `if (!encoder_update_user(index, clockwise)) { return false; }`: I understand this means, by default (user that burns using this default firmware) will get stopped at that line, thus NOT have the encoders behavior?

Reply:
Also, related to your comment @daskygit , would the z12 be listed in the configurator without a info.json file?

Reply:
If the user doesn't include the `encoder_update_user` function in their keymap, the weak function here <https://github.com/qmk/qmk_firmware/blob/master/quantum/encoder.c#L62> is included that returns true. That way the keyboard level code works correctly.

My knowledge of configurator is shamefully lacking but yes you need a info.json for configurator support. I actually thought that was a requirement for a keyboard PR.

Reply:
OK thank you again. I just added the info.json file, had forgotten about it. Sorry.